### PR TITLE
Convert docs from SSR to SSG

### PR DESCRIPTION
**Pull request by ericclemmons**
### Before

	rm -rf docs/.next && time yarn docs build
	✨  Done in 61.51s.
	yarn docs build  218.09s user 36.18s system 412% cpu 1:01.65 total

### After

	✨  Done in 48.16s.
	yarn docs build  141.65s user 44.45s system 385% cpu 48.301 total

*Description of changes:*

- [x] Upgrade `next` from `11.0.1` to `12.0.4`
	
	We couldn't do this before because we had to share the same version of Next.js as https://github.com/aws-amplify/docs

- [x] Opt-in to new Rust/SWC compiler for speed
- [x] Remove `I18n` (that wasn't being used yet), but requires SSR
- [x] Add `next lint` (via `yarn next lint`) to catch any obvious errors
- [x] Fix lint errors (mostly missing `key`)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/next.config.js [hunk 1]
```diff
@@ -12,11 +12,7 @@ module.exports = withNextPluginPreval({
   // Differentiate pages with frontmatter & layout vs. normal MD(X)
   pageExtensions: ['page.mdx', 'page.tsx'],
 
-  // Sets the lang attribute on html element
-  i18n: {
-    locales: ['en'],
-    defaultLocale: 'en',
-  },
+  swcMinify: true,
```
Comment (reesscot) [inline docs/next.config.js:15]: Rust for the win!

File: docs/src/pages/components/card/demo.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ export const CardDemo = () => {
         backgroundColor={tokens.colors.background.secondary}
         padding={tokens.space.medium}
       >
-        <Card {...props}>I'm a card!</Card>
+        <Card {...props}>{`I'm a card!`}</Card>
```
Comment (reesscot) [inline docs/src/pages/components/card/demo.tsx:62]: What does this fix? Was Next adding extra `p` tags?

**Review by reesscot — APPROVED** [review]
LGTM

File: docs/src/pages/components/card/demo.tsx (hunk 2 above)
Comment (ericclemmons) [reply docs/src/pages/components/card/demo.tsx:62]: `next lint` was warning about `'` being an unescaped entity:
> https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md

I've never really had an issue personally...

**Review by dbanksdesign — APPROVED** [review]
LGTM! Let's get this in!

**Comment by ericclemmons** [comment]
I attempted to deploy just this branch and got the following error:

> 2021-11-29T15:17:59 [INFO]: Beginning deployment for application d3inl0muob87le, branch:ssg, buildId 0000000001
2021-11-29T15:17:59 [INFO]: Cannot find any generated SSR resources to deploy. If you intend for your app to be SSR, please check your app Service Role permissions. Otherwise, please check out our docs on how to setup your app to be detected as SSG (https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app)
2021-11-29T15:17:59 [ERROR]: {"code":"7","message":"No ssrResources.json file"}
> 
> – https://us-east-2.console.aws.amazon.com/amplify/home?region=us-east-2&code=ae3ced7b368634a389b9#/d3inl0muob87le/ssg/1?step=DEPLOY

(This sounds related to https://github.com/aws-amplify/amplify-console/issues/2284)

**Comment by ericclemmons** [comment]
Based on https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app, let's see if c0fed3a fixes it...

**Comment by ericclemmons** [comment]
Nope, same error:
> 2021-11-29T15:52:04 [INFO]: Beginning deployment for application d3inl0muob87le, branch:ssg, buildId 0000000002
2021-11-29T15:52:04 [INFO]: Cannot find any generated SSR resources to deploy. If you intend for your app to be SSR, please check your app Service Role permissions. Otherwise, please check out our docs on how to setup your app to be detected as SSG (https://docs.aws.amazon.com/amplify/latest/userguide/server-side-rendering-amplify.html#deploy-nextjs-app)
2021-11-29T15:52:04 [ERROR]: {"code":"7","message":"No ssrResources.json file"}

**Comment by ericclemmons** [comment]
@hvergara recommends we validate the cross-platform behavior when using SSG for:
- https://ui.docs.amplify.aws/components/authenticator?platform=vue
- https://ui.docs.amplify.aws/components/authenticator?platform=angular
- https://ui.docs.amplify.aws/components/authenticator?platform=react

**Comment by ericclemmons** [comment]
@hvergara also recommends validating how googlebot is viewing our pages:
> https://www.google.com/webmasters/tools/googlebot-fetch

**Pull request merged**